### PR TITLE
Feature/distributed ensemble

### PR DESCRIPTION
**Pull request by arbennett**

### CODE REVIEW

File: setup.py [hunk 1]
```diff
@@ -1,25 +1,25 @@
...
+      packages=find_packages(),
+      install_requires=[
+          'numpy',
+          'xarray>0.10.9',
+          'matplotlib',
+          'geopandas',
+          'shapely',
+          'seaborn',
+          'pandas>=0.25',
+          'distributed',
```
Comment (DavidChoi76) [inline setup.py:20]: I think we still need hs_restclient to download and upload HydroShare resources.

File: pysumma/distributed.py [hunk 2]
```diff
@@ -0,0 +1,141 @@
...
+    Currently only supports GRU based parallelization.
+    '''
+
+    def __init__(self, executable: str, filemanager: str,
+                 num_workers: int=1, threads_per_worker: int=OMP_NUM_THREADS,
+                 chunk_size: int=None, num_chunks: int=None, scheduler: str=None):
+        """Initialize a new distributed object"""
+        self._status = 'Initialized'
+        self.executable = executable
+        self.manager_path = Path(os.path.abspath(filemanager))
```
Comment (DavidChoi76) [inline pysumma/distributed.py]: Why don't you change like `self.manager_path = Path(os.path.abspath(os.path.realpath(filemanager)))` as you changed in simulation.py

File: pysumma/distributed.py [hunk 3]
```diff
@@ -0,0 +1,141 @@
...
+    Distributed objects represent SUMMA configurations where
+    there are multiple GRU/HRU which are expected to be run
+    in parallel.
+
+    Currently only supports GRU based parallelization.
+    '''
+
+    def __init__(self, executable: str, filemanager: str,
+                 num_workers: int=1, threads_per_worker: int=OMP_NUM_THREADS,
+                 chunk_size: int=None, num_chunks: int=None, scheduler: str=None):
```
Comment (DavidChoi76) [inline pysumma/distributed.py:28]: Can you explain what the chunk_size and num_chunks are?

File: pysumma/distributed.py (hunk 2 above)
Comment (arbennett) [reply pysumma/distributed.py]: Ah, I should fix that - thanks!

File: pysumma/distributed.py (hunk 3 above)
Comment (arbennett) [reply pysumma/distributed.py:28]: I will add docstrings to explain all of the parameters here.

File: pysumma/distributed.py (hunk 2 above)
Comment (arbennett) [reply pysumma/distributed.py]: Done

File: setup.py (hunk 1 above)
Comment (arbennett) [reply setup.py:20]: Fixed this

File: pysumma/distributed.py (hunk 3 above)
Comment (arbennett) [reply pysumma/distributed.py:28]: Added docstring

**Pull request merged**